index.js

```diff
@@ -1044,10 +1044,44 @@ function getInnerFields(params, fields) {
 function opContainsAnyField(op, fields) {
   for (var i = 0; i < op.length; i++) {
     var component = op[i];
-    if (component.p.length === 0) {
-      return true;
-    } else if (fields[component.p[0]]) {
+    if (Array.isArray(component.p)) {
+      // json0 op component:
+      //
+      // Each op component has its own array of `p` path segments from doc root.
+      if (component.p.length === 0) {
+        return true;
+      } else if (fields[component.p[0]]) {
+        return true;
+      }
+    } else if (typeof component === 'string') {
```

Comment:
This check would break with eg [`text-unicode`](https://github.com/ottypes/text-unicode#spec) with a query set up on the root-level `_data` field (the default field if your snapshot isn't an object)